### PR TITLE
Implement MultiPartStore for InMemory

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Relates to #5458.

# Rationale for this change
 

Makes it easier to use InMemory to test functionality making use of `MultiPartStore`

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: object_store/src/memory.rs [hunk 1]
```diff
@@ -101,6 +108,12 @@ impl Entry {
 struct Storage {
     next_etag: usize,
     map: BTreeMap<Path, Entry>,
+    uploads: HashMap<usize, PartStorage>,
+}
+
+#[derive(Debug, Default, Clone)]
+struct PartStorage {
+    parts: Vec<Option<Bytes>>,
```
Comment (alamb) [inline object_store/src/memory.rs:116]: is it worth nothing in a doc comment somewhere that unfinished multi part uploads are never cleaned up?

**Review by alamb — APPROVED** [review]
LGTM -- thanks @tustvold

File: object_store/src/memory.rs (hunk 1 above)
Comment (tustvold) [reply object_store/src/memory.rs:116]: Oh that's an oversight

Comment (tustvold) [reply object_store/src/memory.rs:116]: I think this is implicit in the contract of MultipartStore

Comment (alamb) [reply object_store/src/memory.rs:116]: ok

**Pull request merged**